File: simplevlcplayer/src/main/java/com/masterwok/simplevlcplayer/dagger/modules/ServiceModule.java
package com.masterwok.simplevlcplayer.dagger.modules;

import com.masterwok.simplevlcplayer.services.MediaPlayerService;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ServiceModule {
    @ContributesAndroidInjector
    abstract MediaPlayerService contributeMediaPlayerService();
}

File: simplevlcplayer/src/main/java/com/masterwok/simplevlcplayer/dagger/modules/ServiceModule.kt
package com.masterwok.simplevlcplayer.dagger.modules

import com.masterwok.opensubtitlesandroid.services.contracts.OpenSubtitlesService
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
@Suppress("unused")
class ServiceModule {

    @Singleton
    @Provides
    fun providesOpenSubtitlesService(): OpenSubtitlesService =
            com.masterwok.opensubtitlesandroid.services.OpenSubtitlesService()
}